src/App.js

```diff
@@ -1,12 +1,218 @@
-import React from 'react';
+import React, {useState} from 'react';
 import './App.css';
+import {useForm} from 'react-hook-form'
+
+import Button from "./components/Button";
+import Section from "./components/Section";
 
 function App() {
-  return (
-    <>
-      <h1>Fruitmand bezorgservice</h1>
-    </>
-  );
+    const {register, handleSubmit, formState: {errors}} = useForm()
+
+    const [state, setState] = useState({
+        strawberry: 0,
+        banana: 0,
+        apple: 0,
+        kiwi: 0,
+    })
+
+    // Destructure state fields
+    const {strawberry, banana, apple, kiwi} = state
+
+    function count(e) {
+        // Check if we need to add or substract
+        const operation = e.target.firstChild.data
+
+        // Check operator and FIRST add or substract, before assigning
+        const newValue = operation === "+" ? ++e.target.value : --e.target.value;
+        setState({
+            ...state,
+            // Make sure not to count below 0
+            [e.target.name]: (newValue < 0) ? 0 : newValue
+        })
+    }
```

Comment:
Wat je misschien de volgende keer zou kunnen doen is een functie maken die een functie teruggeeft. Aan de hand van de parameter van de buitenste functie weet je dan wat de operator is. Voorbeeldje:

```
const doSomething = numberToAdd => e => {
    const newValue = e.target.value + numberToAdd;
    
    if (newValue >= 0){
        setState({
            ...state,
            [e.target.name]: newValue
        })
    }
}

return (
    <>
        <button onClick={doSomething(-1)}>-</button>
        // {...showCounter}
        <button onClick={doSomething(1)}>+</button>
    </>
);
```